pkg/controller/statusmanager/status_manager.go

```diff
@@ -354,6 +356,12 @@ func (status *StatusManager) SetFromPods() {
 	// If all pods are being created, mark deployment as not failing
 	status.SetNotFailing(PodDeployment)
 
+	// If there are deployments or daemonSets created, but no pods
```

Comment:
If we lock the whole getter, we would not need to have this check. Right?

Reply:
If I understand correctly, atomic getter for all 4 attribute would not neccessarily mitigate this.

Reply:
If we have atomic getter and setter, then it can never happen that there are any deployments or daemonsets but no containers. So the following condition would be never true.

Reply:
Added the atomic getter, removed this condition